src/UndoCommands.cpp

```diff
@@ -292,18 +360,35 @@ redo()
 {
   _scene->clearSelection();
 
-  _newSceneJson = takeSceneJsonFromClipboard();
-
-  if (_newSceneJson.empty())
-    return;
-
-  _newSceneJson = makeNewNodeIdsInScene(_newSceneJson);
+  // Ignore if pasted in content that does not generate nodes.
+  try
+  {
+    insertSerializedItems(_newSceneJson, _scene);
+  }
+  catch(...)
```

Comment:
Do you think it might be better to catch a specific exception?

Reply:
I just used the `FlowScene::restoreNode(...)` method in v2, in the `FlowScene.cpp` file. 

With catch I can also catch the error generated by add connections, although it doesn't seem to be there at the moment. This is as advantageous as I can think of at the moment.

Of course I also tried to modify `void loadNode(...)` to `bool`, if it's `true` it means it was loaded successfully. But this can cause quite a lot of changes.